rag/rag-base/src/commonMain/kotlin/ai/koog/rag/base/files/FilteredFileSystemProvider.kt

```diff
@@ -25,6 +33,18 @@ public fun <Path> FileSystemProvider.ReadWrite<Path>.filter(
     return FilteredReadWrite(this, filter)
 }
 
+/**
+ * Filters the current read-write file system implementation such that
+ * only paths that are based on the specified [root] path are visible and accessible.
+ */
+public fun <Path> FileSystemProvider.ReadWrite<Path>.filterByRoot(root: Path): FileSystemProvider.ReadWrite<Path> {
+    return FilteredReadWrite(this, createFilterByRoot(root))
+}
```

Comment:
I'd provide pre-defined filters via PathFilters and let the users combine them as they want